doozerlib/cli/release_gen_payload.py

```diff
@@ -59,762 +146,869 @@ def modify_and_replace_api_object(api_obj: oc.APIObject, modifier_func: Callable
+        exit(1)
 
-    base_imagestream_name: str = is_name if is_name else assembly_imagestream_base_name(runtime)
-    base_istream_namespace: str = is_namespace if is_namespace else default_imagestream_namespace_base_name()
+    def validate_parameters(self):
+        """Sanity check the assembly requested and adjust state accordingly."""
+        assembly = self.runtime.assembly
+        if assembly not in {None, "stream", "test"} and assembly not in self.runtime.releases_config.releases:
+            raise DoozerFatalError(f"Assembly '{assembly}' is not explicitly defined.")
+
+        if assembly and assembly != "stream" and "art-latest" in self.base_imagestream[1]:
+            raise ValueError('"art-latest" imagestreams should only be used for the "stream" assembly')
+
+        if self.runtime.assembly_type is AssemblyTypes.STREAM:
+            # Only nightlies have the concept of private and public payloads
+            self.privacy_modes = [False, True]
+
+    def generate_assembly_report(self, assembly_inspector: AssemblyInspector) -> Dict:
+        """Generate a status report of the search for inconsistencies across all payloads generated."""
+        rt = self.runtime
+        report = dict(
+            non_release_images=[image_meta.distgit_key for image_meta in rt.get_non_release_image_metas()],
+            release_images=[image_meta.distgit_key for image_meta in rt.get_for_release_image_metas()],
+            missing_image_builds=[
+                dgk for (dgk, ii) in assembly_inspector.get_group_release_images().items()
+                if ii is None
+            ],  # A list of metas where the assembly did not find a build
+        )
+        report["viable"], report["assembly_issues"] = self.generate_assembly_issues_report(assembly_inspector)
+
+        self.payload_permitted = report["viable"]
+
+        return report
+
+    def generate_assembly_issues_report(self, assembly_inspector: AssemblyInspector) -> (bool, Dict[str, Dict]):
+        """Populate self.assembly_issues and payload entries with inconsistencies found."""
+        rt = self.runtime
+        self.logger.info("Checking assembly content for inconsistencies.")
+
+        self.detect_mismatched_siblings(assembly_inspector)
+        assembly_build_ids: Set[int] = self.collect_assembly_build_ids(assembly_inspector)
+
+        with rt.shared_build_status_detector() as bsd:
+            # Use the list of builds associated with the group/assembly to warm up BSD caches
+            bsd.populate_archive_lists(assembly_build_ids)
+            bsd.find_shipped_builds(assembly_build_ids)
+
+        # check that RPMs belonging to this assembly/group are consistent with the assembly definition.
+        for rpm_meta in rt.rpm_metas():
+            self.assembly_issues.extend(assembly_inspector.check_group_rpm_package_consistency(rpm_meta))
+        if rt.assembly == "stream":
```

Comment:
Per https://issues.redhat.com/browse/ART-3162, it is better not to hardcode the assembly name:

```suggestion
        self.runtime.assembly_type is AssemblyTypes.STREAM
```